### PR TITLE
Add digitalocean

### LINKED ISSUE #71: Add Digital Ocean installer
This repository contains instructions and configuration for deploying Fermyon locally and on AWS.
It would be great to have Terraform configuration for deploying to Digital Ocean.

ref https://www.digitalocean.com/community/tutorials/how-to-use-terraform-with-digitalocean

### PR DESCRIPTION
**Pull request by FrankYang0529**
resolves #71

### CODE REVIEW

**Review by vdice — COMMENTED** [review]
Thanks @FrankYang0529 for this DO installer!  A great excuse for me to finally try out DO.

Things worked well for me, though did notice a quirk with the default vpc.  Also a few other minor items.

File: digitalocean/README.md [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+(see the configuration details mentioned below). Additionally, when Let's Encrypt
+is enabled, apps will be provided with https URLs and TLS certs courtesy LE.
+
+# Prerequisites
+
+- The [terraform CLI](https://learn.hashicorp.com/tutorials/terraform/install-cli#install-terraform)
+
+# Resources deployed
+
+This example creates the following resources in the provided GCP project:
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/README.md]: Fixed it. Thank you.

File: digitalocean/README.md [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+As this example takes a stock Ubuntu 20.04 LTS and then proceeds to download Fermyon and Hashistack binaries,
+the default inbound CIDR block is likely necessary for first startup. After confirmation that
+Fermyon is up and running - and as long as subsequent apps/workloads won't need access to
+the broader internet - this value may be updated on a subsequent `terraform apply` if desired, e.g.
+`terraform apply -var=allowed_inbound_cidr_blocks=["75.75.75.75/32"]`.
+
+# How to Deploy
+
+First, [create a personal access token](https://docs.digitalocean.com/reference/api/create-personal-access-token/) and set `DIGITALOCEAN_TOKEN` environment variable.
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/README.md]: Updated it. Thank you.

File: digitalocean/README.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+```console
+cd terraform
+terraform init
+```
+
+Deploy with all defaults (http-based URLs):
+
+```console
+terraform apply'
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/README.md]: Fixed it. Thank you.

File: digitalocean/README.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+without any redundancy on a single Droplet instance. There is no data backup for any
+service.
+
+That being said, it should give users a quick look and feel for deploying apps
+using Fermyon. By default, all apps will be accessible to the broader internet
+(see the configuration details mentioned below). Additionally, when Let's Encrypt
+is enabled, apps will be provided with https URLs and TLS certs courtesy LE.
+
+# Prerequisites
+
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/README.md:18]: Added account and personal access token in the prerequisites section.

File: digitalocean/README.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+Deploy with all defaults and use Let's Encrypt to provision certs for TLS/https:
+
+```console
+terraform apply -var='enable_letsencrypt=true'
+```
+
+Deploy with a custom instance name, perhaps so multiple examples can co-exist in the same region:
+
+```console
+terraform apply -var='instance_name=fermyonrocks'
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/README.md]: Fixed it. Thank you.

File: digitalocean/terraform/main.tf [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+
+locals {
+  dependencies = yamldecode(file("../../share/terraform/dependencies.yaml"))
+}
+
+# -----------------------------------------------------------------------------
+# Default VPC
+# -----------------------------------------------------------------------------
+
+resource "digitalocean_vpc" "default" {
```
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/terraform/main.tf:13]: I tried to add `lifecycle` like the following, but it doesn't work.

```
lifecycle {
    prevent_destroy = self.default ? false : true
  }
```

I also find an related issue https://github.com/digitalocean/terraform-provider-digitalocean/issues/472#issuecomment-668094991.

**Comment by vdice** [comment]
CI failure should be fixed in https://github.com/fermyon/installer/pull/87 -- @FrankYang0529 can you rebase off of `main` when convenient?  Sorry for the extra trouble.

File: digitalocean/terraform/main.tf (hunk 6 above)
[the remark this answers is not in the document]
Comment (FrankYang0529) [reply digitalocean/terraform/main.tf:13]: I add a note for it in the Resource deployed section. Please help me check again. Thank you.

**Review by vdice — APPROVED** [review]
LGTM, thank you for adding this DigitalOcean installer @FrankYang0529 ! 🚀 🌊

**Comment by FrankYang0529** [comment]
Updated it. Thank you!

**Pull request merged**